nixos/modules/services/networking/dnsmasq.md

```diff
@@ -0,0 +1,62 @@
+# Dnsmasq {#module-services-networking-dnsmasq}
+
+Dnsmasq is integrated DNS, DHCP and TFTP server for small networks.
+
+## Configuration {#module-services-networking-dnsmasq-configuration}
+
+### An authoritative DHCP and DNS server on a home network {#module-services-networking-dnsmasq-configuration-home}
+
+On a home network, you can use Dnsmasq as a DHCP and DNS server. New devices on
+your network will be configured by Dnsmasq, and instructed to use it as the DNS
+server by default. This allows you to rely on your own server to perform DNS
+queries and caching, with DNSSEC enabled.
+
+The following example assumes that
+
+- you have disabled your router's integrated DHCP server, if it has one
+- your router's address is set in  [](#opt-networking.defaultGateway.address)
+- your system's Ethernet interface is `eth0`
+- you have configured the address(es) to forward DNS queries in [](#opt-networking.nameservers)
+
+```nix
```

Comment:
you only explain 2 or 3 out of many options, can you either add more comments or reduce the example to the strict minimum?
Just to name one: I have no idea what bogus-priv is, and this name is just weird.

Reply:
Added more comments, mostly from the man page. All of these seem to be recommended, at least by the Pi-hole project. The more obscure ones are for preventing undesirable requests being sent to upstream DNS nameservers

Reply:
thank you